backend/metering_billing/views/model_views.py

```diff
@@ -426,6 +427,19 @@ class EventViewSet(
     def get_queryset(self):
         now = now_utc()
         organization = self.request.organization
+        distinct = self.request.query_params.get("distinct")
+        if distinct:
+            events = Event.objects.filter(
+                organization=organization, time_created__lt=now
+            )
+            distinct_names = []
+            distinct_uuids = []
+            for event in events.all():
```

Comment:
Evaluating this queryset could be super inefficient as its a table with potentially millions of rows. Ideally want to keep as much of it in the database as we can.

Reply:
I think we could filter by pattern at the db level first and then evaluate. Thoughts @diego-escobedo ?

Reply:
yep, not in this method, but the approach would be to take in an event name + the property filters and filter at db level, and only after that return